lines.go

```diff
@@ -27,14 +27,24 @@ func newLine(line *bill.Line) *Line {
 		Measure:       string(line.Item.Unit.UNECE()),
 		NetUnitPrice:  line.Item.Price.String(),
 		Quantity:      line.Quantity.String(),
-		UnitDiscount:  line.Total.Subtract(line.Sum).Divide(line.Quantity).String(), // not sure if there should be some rescale or matchPrecision
+		UnitDiscount:  unitDiscount(line),
 		NetPriceTotal: line.Total.String(),
 		TaxRate:       line.Taxes[0].Percent.Rescale(2).StringWithoutSymbol(),
 	}
 
 	return Line
 }
 
+func unitDiscount(line *bill.Line) string {
+	discount := line.Total.Subtract(line.Sum).Divide(line.Quantity) // not sure if there should be some rescale or matchPrecision
+
+	if discount.IsZero() {
+		return ""
+	}
+
+	return discount.String()
```

Comment:
What type of discount is it? I ask because since we perform mathematical operations (subtraction and division), these calculations are probably sensitive to precision.

If I understand correctly, the `discount` will be 0 if `line.Total` is equal to `line.Sum`. We can check this at the very beginning of the function.

```suggestion
	if line.Total == line.Sum {
		return ""
	}

	discount := line.Total.Subtract(line.Sum).Divide(line.Quantity) // not sure if there should be some rescale or matchPrecision

	return discount.String()
```
What do you think?

Reply:
Yes, this is a better approach. 👍